alpha/apps/kaltura/lib/kEntryPlayingDataHelper.php

```diff
@@ -0,0 +1,443 @@
+		if (!count($remoteFlavorsByDc))
+			return $sources;
+
+		$deliveryAttributes = DeliveryProfileDynamicAttributes::init(null, $entryId, null);
+		$remoteDeliveryProfiles = DeliveryProfilePeer::getDeliveriesByIds($dbEntry, $remoteDeliveryProfileIds, $dbEntry->getPartner(), $deliveryAttributes);
+
+		list($deliveryProfileIds, $deliveryProfilesParamsNotIn) = $this->getProfileIdsToFilter($contextDataHelper);
+		if (count($deliveryProfileIds))
+			$this->filterDeliveryProfiles($remoteDeliveryProfiles, $deliveryProfileIds, $deliveryProfilesParamsNotIn);
+
+		foreach ($remoteDeliveryProfiles as $deliveryProfile)
+		{
+			$dcFlavorIds = array();
+			$dc = $remoteDcByDeliveryProfile[$deliveryProfile->getId()];
+			$flavorAssetsForDc = $remoteFlavorsByDc[$dc];
+
+			if (count($flavorAssetsForDc))
+			{
+				$deliveryProfileFlavorsForDc = $flavorAssetsForDc;
+
+				$flavorToDrmData = $this->getDrmData($dbEntry, $deliveryProfileFlavorsForDc, $deliveryProfile);
+
+				if (count($deliveryProfileFlavorsForDc))
+				{
+					foreach ($deliveryProfileFlavorsForDc as $flavorAssetForDc)
+						$dcFlavorIds[] = $flavorAssetForDc->getId();
+					$sources[] = new KalturaPlayingSource($deliveryProfile->getId(), $deliveryProfile->getStreamerType(), $deliveryProfile->getPriority(), $this->constructProtocol($deliveryProfile), $dcFlavorIds, null, $flavorToDrmData);
+				}
+			}
+		}
+
+		return $sources;
+	}
+
+	/**
+	 * @param $flavorAssets
+	 * @param $flavorAssetsIds
+	 * @param $flavorAssetsParamsNotIn
+	 */
+	protected function filterFlavorAssets(&$flavorAssets, $flavorAssetsIds, $flavorAssetsParamsNotIn)
+	{
+		foreach ($flavorAssets as $key => $flavorAsset)
+		{
+			if (in_array($flavorAsset->getId(), $flavorAssetsIds))
+			{
+				if ($flavorAssetsParamsNotIn)
+					unset($flavorAssets[$key]);
+			} else
+			{
+				if (!$flavorAssetsParamsNotIn)
+					unset($flavorAssets[$key]);
+			}
+		}
+	}
+
+	/**
+	 * @param $deliveryProfiles
+	 * @param $deliveryProfileIds
+	 * @param $deliveryProfilesParamsNotIn
+	 */
+	protected function filterDeliveryProfiles(&$deliveryProfiles, $deliveryProfileIds, $deliveryProfilesParamsNotIn)
+	{
+		foreach ($deliveryProfiles as $key => $deliveryProfile)
+		{
+			if (in_array($deliveryProfile->getId(), $deliveryProfileIds))
+			{
+				if ($deliveryProfilesParamsNotIn)
+				{
+					unset($deliveryProfiles[$key]);
+				}
+			} else
+			{
+				if (!$deliveryProfilesParamsNotIn)
+					unset($deliveryProfiles[$key]);
+			}
+		}
+	}
+
+	protected function getStreamsTypeToExclude($localCustomDeliveries)
+	{
+		$streamTypes = array();
+		foreach ($localCustomDeliveries as $deliveryProfile)
+			$streamTypes[] = $deliveryProfile->getStreamerType();
+
+		return $streamTypes;
+	}
+
+	/* @param $dbEntry
+	 * @param $flavorAssets
+	 * @param $deliveryProfile
+	 * @return KalturaEntryPlayingDataResult
+	 */
+	protected function getDrmData($dbEntry, &$flavorAssets, $deliveryProfile)
+	{
+		$playingDataParams = new KalturaEntryPlayingDataParams();
```

Comment:
KalturaEntryPlayingDataParams, KalturaFlavorAssetArray, KalturaEntryPlayingDataResult, KalturaPluginDataArray are all API objects and shouldn't be used in a core class.